### PR TITLE
[WIP] Add demo project to test CocoaPods integration

### PR DESCRIPTION
**Pull request by sharplet**
There's also some minor improvements we could make to the main project file, and I'd like to add a target to build and test with CocoaPods on CI. Watch this space.

### CODE REVIEW

File: Demo/Podfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+target "CocoaPodsTests-iOS" do
+  platform :ios, "8.0"
+  pod "Kiwi", path: ".."
+end
+
+target "CocoaPodsTests-OSX" do
+  platform :osx, "10.10"
+  pod "Kiwi", path: ".."
+end
```
Comment (sharplet) [inline Demo/Podfile]: Is it worth add a `Gemfile` so that we can have some determinism around the CocoaPods version we've tested with? @supermarin

Comment (supermarin) [inline Demo/Podfile]: @sharplet this looks like a `Podfile`, not `Gemfile` :)

Comment (supermarin) [inline Demo/Podfile]: Version of CocoaPods is always stored in Podfile.lock.
I'd go with the lighter approach here and just try this way.

Comment (sharplet) [inline Demo/Podfile]: Yep, agreed.

**Comment by orta** [comment]
I don't think this is needed, the `pod lint` aspects of - https://github.com/kiwi-bdd/Kiwi/blob/master/Makefile#L35-L39 will ( with modern versions of CocoaPods ( I think 0.39 does this )) will do all that is being replicated in this PR

**Pull request closed without merging**